### PR TITLE
initial nessus integration

### PR DESCRIPTION
**Pull request by sfowl**
Early PR just to demo initial approach.

### CODE REVIEW

File: scanners/nessus/nessus_none.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    "aborted",
+]
+
+
+class Nessus(RapidastScanner):
+    def __init__(self, config: RapidastConfigModel, ident: str = "nessus"):
+        super().__init__(config, ident)
+        self._nessus_client: Optional[PyNessusPro] = None
+        self._scan_id: Optional[int] = None
+        nessus_config_section = config.subtree_to_dict("scanners.nessus")
```
[the remark this answers is not in the document]
Comment (sfowl) [reply scanners/nessus/nessus_none.py]: Ah I see, there's some magic in scanners/__init__.py that will match sections to sections based on prefix, separated with underscores. Should be fixed now.

File: scanners/nessus/nessus_none.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        if nessus_config_section is None:
+            raise ValueError("'scanners.nessus' section not in config")
+        # self.auth_config = config.subtree_to_dict("general.authentication")  # creds for target hosts
+        self.config = dacite.from_dict(data_class=NessusConfig, data=nessus_config_section)
```
[the remark this answers is not in the document]
Comment (sfowl) [reply scanners/nessus/nessus_none.py]: Renamed `config` -> `cfg`

**Review by cedricbu — APPROVED** [review]
Looks good to me!

File: scanners/nessus/nessus_none.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    @property
+    def scan_id(self) -> int:
+        if self._scan_id is None:
+            raise RuntimeError("scan_id is None")
+        return self._scan_id
+
+    def setup(self):
+        logging.debug(f"Creating new scan named {self.cfg.scan.folder}/{self.cfg.scan.name}")
+        self._scan_id = self.nessus_client.new_scan(
```
Comment (jeremychoi) [inline scanners/nessus/nessus_none.py:93]: any chance new_scan() could fail and return -1(https://github.com/Matbe34/py-nessus-pro/blob/f49a60643c7d553ae9ae8172f6454028fafc852e/py_nessus_pro/py_nessus_pro.py#L100)? then handling the exception would be necessary.

File: scanners/nessus/nessus_none.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        logging.debug("Launching scan")
+        self.nessus_client.set_scan_launch_now(scan_id=self.scan_id, launch_now=True)
+
+        # Tell nessus to create and launch the scan
+        self.nessus_client.post_scan(scan_id=self.scan_id)
+
+        # Wait for the scan to complete
+        while self.nessus_client.get_scan_status(self.scan_id)["status"] not in END_STATUSES:
+            time.sleep(self.sleep_interval)
+            logging.debug("Waiting {self.sleep_interval}s for scan to finish")
```
Comment (jeremychoi) [inline scanners/nessus/nessus_none.py:121]: f"Waiting ..."

File: scanners/nessus/nessus_none.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            logging.info(self.nessus_client.get_scan_status(self.scan_id))
+
+    def postprocess(self):
+        # After scan is complete, download report in csv, nessus, and html format
+        # Path and any folders must already exist in this implementation
+        logging.debug("Retrieving scan reports")
+        scan_reports = self.nessus_client.get_scan_reports(self.scan_id, self.results_dir)
+        print(scan_reports)
```
Comment (jeremychoi) [inline scanners/nessus/nessus_none.py:129]: We may want to use logging instead of print() for consistency?

File: scanners/nessus/nessus_none.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            raise RuntimeError(f"[nessus] unexpected state: READY != {self.state}")
+        # State that we want the scan to launch immediately
+        logging.debug("Launching scan")
+        self.nessus_client.set_scan_launch_now(scan_id=self.scan_id, launch_now=True)
+
+        # Tell nessus to create and launch the scan
+        self.nessus_client.post_scan(scan_id=self.scan_id)
+
+        # Wait for the scan to complete
+        while self.nessus_client.get_scan_status(self.scan_id)["status"] not in END_STATUSES:
```
Comment (jeremychoi) [inline scanners/nessus/nessus_none.py:119]: any chance a scan doesn't end?

File: scanners/nessus/nessus_none.py (hunk 4 above)
Comment (sfowl) [reply scanners/nessus/nessus_none.py]: fixed

File: scanners/nessus/nessus_none.py (hunk 6 above)
Comment (sfowl) [reply scanners/nessus/nessus_none.py:124]: added a timeout, default 300 seconds

File: scanners/nessus/nessus_none.py (hunk 5 above)
Comment (sfowl) [reply scanners/nessus/nessus_none.py]: switched to logging.debug

File: scanners/nessus/nessus_none.py (hunk 3 above)
Comment (sfowl) [reply scanners/nessus/nessus_none.py:94]: Not 100% sure if I understood correctly, but I added an error check which will raise an exception if the scan_id is less than 0.

**Review by jeremychoi — APPROVED** [review]
LGTM

**Pull request merged**